src/main/java/org/kcctl/command/GetPluginsCommand.java

```diff
@@ -51,19 +61,35 @@ public GetPluginsCommand() {
     }
 
     @Override
-    public void run() {
+    public Integer call() {
         KafkaConnectApi kafkaConnectApi = RestClientBuilder.newBuilder()
                 .baseUri(context.getCurrentContext().getCluster())
                 .build(KafkaConnectApi.class);
 
-        List<ConnectorPlugin> connectorPlugins = kafkaConnectApi.getConnectorPlugins();
-        connectorPlugins.sort((c1, c2) -> -c1.type().compareTo(c2.type()));
+        Version currentVersion = new Version(kafkaConnectApi.getWorkerInfo().version());
+
+        if (pluginTypes != null) {
+            if (!currentVersion.greaterOrEquals(requiredVersionForAllPlugins)) {
+                spec.commandLine().getErr().println("--types requires at least Kafka Connect 3.2. Current version: " + currentVersion);
+                return 1;
+            }
```

Comment:
We could apply client-side filtering of the result set for KC < 3.2. We have similar logic elsewhere already I think, kinda emulating features when absent in older versions and doable with a reasonable effort.

Reply:
The issue is that pre 3.2 you cannot list all plugins. I've updated the logic to return an error in case you try to list plugins other than sink and source without 3.2.

For example:
```
> kcctl get plugins --types=predicate
Listing plugins other than source and sink requires at least Kafka Connect 3.2. Current version: 3.1
> kcctl get plugins --types=sink

 TYPE   CLASS                                                   VERSION
 sink   org.apache.kafka.connect.file.FileStreamSinkConnector   3.1.0
```